RabbitMQ properties is getting in the middle when Kafka is used as binder

There seems to be something referencing: org.springframework.cloud.stream.binder.rabbit.properties.RabbitProducerProperties from the connector or RuntimeBundle starter projects. 

```
2019-01-22 09:21:18.484 ERROR [provisioning-rb,,,] 17752 --- [  restartedMain] o.s.boot.SpringApplication               : Application run failed

java.lang.IllegalStateException: Error processing condition on org.conf.activiti.runtime.api.ProcessRuntimeAutoConfiguration.signalPayloadEventListener
    at org.springframework.boot.autoconfigure.condition.SpringBootCondition.matches(SpringBootCondition.java:64) ~[spring-boot-autoconfigure-2.1.0.RELEASE.jar:2.1.0.RELEASE]
    at org.springframework.context.annotation.ConditionEvaluator.shouldSkip(ConditionEvaluator.java:108) ~[spring-context-5.1.2.RELEASE.jar:5.1.2.RELEASE]
    at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.loadBeanDefinitionsForBeanMethod(ConfigurationClassBeanDefinitionReader.java:181) ~[spring-context-5.1.2.RELEASE.jar:5.1.2.RELEASE]
    at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.loadBeanDefinitionsForConfigurationClass(ConfigurationClassBeanDefinitionReader.java:141) ~[spring-context-5.1.2.RELEASE.jar:5.1.2.RELEASE]
    at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.loadBeanDefinitions(ConfigurationClassBeanDefinitionReader.java:117) ~[spring-context-5.1.2.RELEASE.jar:5.1.2.RELEASE]
    at org.springframework.context.annotation.ConfigurationClassPostProcessor.processConfigBeanDefinitions(ConfigurationClassPostProcessor.java:327) ~[spring-context-5.1.2.RELEASE.jar:5.1.2.RELEASE]
    at org.springframework.context.annotation.ConfigurationClassPostProcessor.postProcessBeanDefinitionRegistry(ConfigurationClassPostProcessor.java:232) ~[spring-context-5.1.2.RELEASE.jar:5.1.2.RELEASE]
    at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanDefinitionRegistryPostProcessors(PostProcessorRegistrationDelegate.java:275) ~[spring-context-5.1.2.RELEASE.jar:5.1.2.RELEASE]
    at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:95) ~[spring-context-5.1.2.RELEASE.jar:5.1.2.RELEASE]
    at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:691) ~[spring-context-5.1.2.RELEASE.jar:5.1.2.RELEASE]
    at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:528) ~[spring-context-5.1.2.RELEASE.jar:5.1.2.RELEASE]
    at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:140) ~[spring-boot-2.1.0.RELEASE.jar:2.1.0.RELEASE]
    at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:775) [spring-boot-2.1.0.RELEASE.jar:2.1.0.RELEASE]
    at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:397) [spring-boot-2.1.0.RELEASE.jar:2.1.0.RELEASE]
    at org.springframework.boot.SpringApplication.run(SpringApplication.java:316) [spring-boot-2.1.0.RELEASE.jar:2.1.0.RELEASE]
    at org.springframework.boot.SpringApplication.run(SpringApplication.java:1260) [spring-boot-2.1.0.RELEASE.jar:2.1.0.RELEASE]
    at org.springframework.boot.SpringApplication.run(SpringApplication.java:1248) [spring-boot-2.1.0.RELEASE.jar:2.1.0.RELEASE]
    at io.rasolog.provisioning.ProvisioningRuntimeBundleApplication.main(ProvisioningRuntimeBundleApplication.java:15) [classes/:na]
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_202-ea]
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_202-ea]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_202-ea]
    at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_202-ea]
    at org.springframework.boot.devtools.restart.RestartLauncher.run(RestartLauncher.java:49) [spring-boot-devtools-2.1.0.RELEASE.jar:2.1.0.RELEASE]
Caused by: java.lang.IllegalStateException: Failed to introspect Class [org.conf.activiti.services.connectors.CloudConnectorsAutoConfiguration] from ClassLoader [sun.misc.Launcher$AppClassLoader@18b4aac2]
    at org.springframework.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:680) ~[spring-core-5.1.2.RELEASE.jar:5.1.2.RELEASE]
    at org.springframework.util.ReflectionUtils.doWithMethods(ReflectionUtils.java:577) ~[spring-core-5.1.2.RELEASE.jar:5.1.2.RELEASE]
    at org.springframework.util.ReflectionUtils.doWithMethods(ReflectionUtils.java:562) ~[spring-core-5.1.2.RELEASE.jar:5.1.2.RELEASE]
    at org.springframework.util.ReflectionUtils.getUniqueDeclaredMethods(ReflectionUtils.java:620) ~[spring-core-5.1.2.RELEASE.jar:5.1.2.RELEASE]
    at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660) ~[na:1.8.0_202-ea]
    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.getTypeForFactoryMethod(AbstractAutowireCapableBeanFactory.java:721) ~[spring-beans-5.1.2.RELEASE.jar:5.1.2.RELEASE]
    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.determineTargetType(AbstractAutowireCapableBeanFactory.java:662) ~[spring-beans-5.1.2.RELEASE.jar:5.1.2.RELEASE]
    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.predictBeanType(AbstractAutowireCapableBeanFactory.java:630) ~[spring-beans-5.1.2.RELEASE.jar:5.1.2.RELEASE]
    at org.springframework.beans.factory.support.AbstractBeanFactory.isFactoryBean(AbstractBeanFactory.java:1491) ~[spring-beans-5.1.2.RELEASE.jar:5.1.2.RELEASE]
    at org.springframework.beans.factory.support.AbstractBeanFactory.isFactoryBean(AbstractBeanFactory.java:1014) ~[spring-beans-5.1.2.RELEASE.jar:5.1.2.RELEASE]
    at org.springframework.boot.autoconfigure.condition.BeanTypeRegistry.addBeanTypeForNonAliasDefinition(BeanTypeRegistry.java:195) ~[spring-boot-autoconfigure-2.1.0.RELEASE.jar:2.1.0.RELEASE]
    at org.springframework.boot.autoconfigure.condition.BeanTypeRegistry.addBeanTypeForNonAliasDefinition(BeanTypeRegistry.java:159) ~[spring-boot-autoconfigure-2.1.0.RELEASE.jar:2.1.0.RELEASE]
    at org.springframework.boot.autoconfigure.condition.BeanTypeRegistry.addBeanType(BeanTypeRegistry.java:152) ~[spring-boot-autoconfigure-2.1.0.RELEASE.jar:2.1.0.RELEASE]
    at org.springframework.boot.autoconfigure.condition.BeanTypeRegistry.updateTypesIfNecessary(BeanTypeRegistry.java:140) ~[spring-boot-autoconfigure-2.1.0.RELEASE.jar:2.1.0.RELEASE]
    at java.util.Iterator.forEachRemaining(Iterator.java:116) ~[na:1.8.0_202-ea]
    at org.springframework.boot.autoconfigure.condition.BeanTypeRegistry.updateTypesIfNecessary(BeanTypeRegistry.java:135) ~[spring-boot-autoconfigure-2.1.0.RELEASE.jar:2.1.0.RELEASE]
    at org.springframework.boot.autoconfigure.condition.BeanTypeRegistry.getNamesForType(BeanTypeRegistry.java:97) ~[spring-boot-autoconfigure-2.1.0.RELEASE.jar:2.1.0.RELEASE]
    ... 22 common frames omitted
Caused by: java.lang.NoClassDefFoundError: org/springframework/cloud/stream/binder/rabbit/properties/RabbitProducerProperties
    at java.lang.Class.getDeclaredMethods0(Native Method) ~[na:1.8.0_202-ea]
    at java.lang.Class.privateGetDeclaredMethods(Class.java:2701) ~[na:1.8.0_202-ea]
    at java.lang.Class.getDeclaredMethods(Class.java:1975) ~[na:1.8.0_202-ea]
    at org.springframework.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:662) ~[spring-core-5.1.2.RELEASE.jar:5.1.2.RELEASE]
    ... 44 common frames omitted
Caused by: java.lang.ClassNotFoundException: org.springframework.cloud.stream.binder.rabbit.properties.RabbitProducerProperties
    at java.net.URLClassLoader.findClass(URLClassLoader.java:382) ~[na:1.8.0_202-ea]
    at java.lang.ClassLoader.loadClass(ClassLoader.java:424) ~[na:1.8.0_202-ea]
    at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349) ~[na:1.8.0_202-ea]
    at java.lang.ClassLoader.loadClass(ClassLoader.java:357) ~[na:1.8.0_202-ea]
    ... 48 common frames omitted
```

--- a/activiti-cloud-services-runtime-bundle/activiti-cloud-services-connectors/pom.xml
+++ b/activiti-cloud-services-runtime-bundle/activiti-cloud-services-connectors/pom.xml
@@ -49,6 +49,7 @@
     <dependency>
       <groupId>org.springframework.cloud</groupId>
       <artifactId>spring-cloud-starter-stream-rabbit</artifactId>
+      <optional>true</optional>
     </dependency>
     <dependency>
       <groupId>org.springframework.data</groupId>

--- a/activiti-cloud-services-runtime-bundle/activiti-cloud-services-connectors/src/main/java/org/conf/activiti/services/connectors/CloudConnectorsAutoConfiguration.java
+++ b/activiti-cloud-services-runtime-bundle/activiti-cloud-services-connectors/src/main/java/org/conf/activiti/services/connectors/CloudConnectorsAutoConfiguration.java
@@ -26,43 +26,21 @@ import org.activiti.runtime.api.connector.VariablesMatchHelper;
 import org.activiti.services.connectors.behavior.MQServiceTaskBehavior;
 import org.activiti.services.connectors.message.IntegrationContextMessageBuilderFactory;
 import org.conf.activiti.runtime.api.ConnectorsAutoConfiguration;
-import org.springframework.beans.factory.annotation.Value;
 import org.springframework.boot.autoconfigure.AutoConfigureBefore;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
-import org.springframework.cloud.stream.binder.rabbit.properties.RabbitProducerProperties;
-import org.springframework.cloud.stream.binding.BinderAwareChannelResolver.NewDestinationBindingCallback;
 import org.springframework.context.ApplicationContext;
 import org.springframework.context.ApplicationEventPublisher;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.ComponentScan;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.PropertySource;
-import org.springframework.expression.Expression;
-import org.springframework.expression.spel.standard.SpelExpressionParser;
 
 @Configuration
 @AutoConfigureBefore(value = ConnectorsAutoConfiguration.class)
-@PropertySource("classpath:config/integration-result-stream.properties")
 @ComponentScan("org.activiti.core.common.spring.connector")
+@PropertySource("classpath:config/integration-result-stream.properties")
 public class CloudConnectorsAutoConfiguration {
 
-    @Value("${activiti.spring.cloud.stream.connector.integrationRequestSender.routing-key-expression}")
-    private String routingKeyExpression;
-
-    /**
-     * Configures routing key expression for dynamic cloud connector destinations if rabbit binder exists
-     */
-    @Bean
-    @ConditionalOnClass(RabbitProducerProperties.class)
-    @ConditionalOnMissingBean
-    public NewDestinationBindingCallback<RabbitProducerProperties> dynamicConnectorDestinationsBindingCallback() {
-        return (channelName, channel, producerProperties, extendedProducerProperties) -> {
-            Expression expression = new SpelExpressionParser().parseExpression(routingKeyExpression);
-            
-            extendedProducerProperties.setRoutingKeyExpression(expression);
-        };
-    }
     
     @Bean
     @ConditionalOnMissingBean

--- a/activiti-cloud-services-runtime-bundle/activiti-cloud-services-connectors/src/main/java/org/conf/activiti/services/connectors/RabbitCloudConnectorsAutoConfiguration.java
+++ b/activiti-cloud-services-runtime-bundle/activiti-cloud-services-connectors/src/main/java/org/conf/activiti/services/connectors/RabbitCloudConnectorsAutoConfiguration.java
@@ -1,0 +1,34 @@
+package org.conf.activiti.services.connectors;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.cloud.stream.binder.rabbit.properties.RabbitProducerProperties;
+import org.springframework.cloud.stream.binding.BinderAwareChannelResolver.NewDestinationBindingCallback;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.expression.Expression;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+
+/**
+ * Configures routing key expression for dynamic cloud connector destinations if rabbit binder exists on classpath
+ */
+@Configuration
+@ConditionalOnClass(RabbitProducerProperties.class)
+@AutoConfigureAfter(CloudConnectorsAutoConfiguration.class)
+public class RabbitCloudConnectorsAutoConfiguration {
+
+    @Value("${activiti.spring.cloud.stream.connector.integrationRequestSender.routing-key-expression:headers['routingKey']}")
+    private String routingKeyExpression;
+    
+    @Bean
+    @ConditionalOnMissingBean
+    public NewDestinationBindingCallback<RabbitProducerProperties> dynamicConnectorDestinationsBindingCallback() {
+        return (channelName, channel, producerProperties, extendedProducerProperties) -> {
+            Expression expression = new SpelExpressionParser().parseExpression(routingKeyExpression);
+            
+            extendedProducerProperties.setRoutingKeyExpression(expression);
+        };
+    }
+}

--- a/activiti-cloud-services-runtime-bundle/activiti-cloud-services-connectors/src/main/resources/META-INF/spring.factories
+++ b/activiti-cloud-services-runtime-bundle/activiti-cloud-services-connectors/src/main/resources/META-INF/spring.factories
@@ -1,2 +1,3 @@
 org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
-    org.conf.activiti.services.connectors.CloudConnectorsAutoConfiguration
+    org.conf.activiti.services.connectors.CloudConnectorsAutoConfiguration,\
+	org.conf.activiti.services.connectors.RabbitCloudConnectorsAutoConfiguration

--- a/activiti-cloud-services-runtime-bundle/activiti-cloud-services-events/pom.xml
+++ b/activiti-cloud-services-runtime-bundle/activiti-cloud-services-events/pom.xml
@@ -100,10 +100,6 @@
       <artifactId>spring-cloud-stream</artifactId>
     </dependency>
     <dependency>
-      <groupId>org.springframework.cloud</groupId>
-      <artifactId>spring-cloud-starter-stream-rabbit</artifactId>
-    </dependency>
-    <dependency>
       <groupId>org.slf4j</groupId>
       <artifactId>slf4j-api</artifactId>
     </dependency>

--- a/activiti-cloud-services-runtime-bundle/activiti-cloud-services-subscriptions/pom.xml
+++ b/activiti-cloud-services-runtime-bundle/activiti-cloud-services-subscriptions/pom.xml
@@ -58,10 +58,6 @@
       <artifactId>spring-cloud-stream</artifactId>
     </dependency>
     <dependency>
-      <groupId>org.springframework.cloud</groupId>
-      <artifactId>spring-cloud-starter-stream-rabbit</artifactId>
-    </dependency>
-    <dependency>
       <groupId>org.springframework.data</groupId>
       <artifactId>spring-data-commons</artifactId>
     </dependency>

--- a/activiti-cloud-starter-runtime-bundle/pom.xml
+++ b/activiti-cloud-starter-runtime-bundle/pom.xml
@@ -70,6 +70,10 @@
       <artifactId>jackson-annotations</artifactId>
     </dependency>
     <dependency>
+      <groupId>org.springframework.cloud</groupId>
+      <artifactId>spring-cloud-starter-stream-rabbit</artifactId>
+    </dependency>    
+    <dependency>
       <groupId>org.springframework.boot</groupId>
       <artifactId>spring-boot-starter-jdbc</artifactId>
     </dependency>

--- a/activiti-cloud-starter-runtime-bundle/pom.xml
+++ b/activiti-cloud-starter-runtime-bundle/pom.xml
@@ -72,7 +72,6 @@
     <dependency>
       <groupId>org.springframework.cloud</groupId>
       <artifactId>spring-cloud-starter-stream-rabbit</artifactId>
-      <optional>true</optional>
     </dependency>    
     <dependency>
       <groupId>org.springframework.boot</groupId>

--- a/activiti-cloud-starter-runtime-bundle/pom.xml
+++ b/activiti-cloud-starter-runtime-bundle/pom.xml
@@ -72,6 +72,7 @@
     <dependency>
       <groupId>org.springframework.cloud</groupId>
       <artifactId>spring-cloud-starter-stream-rabbit</artifactId>
+      <optional>true</optional>
     </dependency>    
     <dependency>
       <groupId>org.springframework.boot</groupId>

